routes/admin.js

```diff
@@ -0,0 +1,280 @@
+const keys = require("../config/nodeKeys");
+const { spawn } = require('child_process');
+const passport = require("passport");
+const flash = require("express-flash");
+const session = require("express-session");
+const path = require('path');
+const bcrypt = require('bcrypt');
+
+module.exports = (app, pool) => {
+
+  /* Configure Passport, the login mechanism for the admin page */
+  const initializePassport = require("../initializePassport");
+  initializePassport(passport, pool);
+  app.use(
+    session({
+      secret: 'secret',
+      resave: false,
+      saveUninitialized: false
+    })
+  );
+  app.use(passport.initialize());
+  app.use(passport.session());
+  app.use(flash());
+  app.use(function (req, res, next) {
+    res.locals.error = req.flash("error");
+    res.locals.success = req.flash("success")
+    next();
+  });
+
+  /* Default handler for the admin page */
+  app.get("/admin/dashboard", userIsAuthenticated, async (req, res, next) => {
+    try {
+
+      const { action } = req.query;
+
+      if (action === 'runetl') {
+        /* The 'Import to Staging' button was clicked */
+
+        /* Prepare to run the ETL script */
+        await clearTables().catch(e => console.log(e));
+        log('Job Start');
+
+        /* Run the ETL script */
+        const file = path.resolve('ETL/main.py');
+        const python = spawn('python3', [file, keys.PG_CONNECTION_STRING]);
+        python.on('spawn', (code) => {
+          console.log('Python spawn: ' + code)
+        })
+        python.on('error', (err) => {
+          console.log('Python error: ' + err)
+        })
+        python.on('exit', (code) => {
+          console.log('Python exit code: ' + code)
+        })
+        python.stderr.on('data', (data) => {
+          log('Python stderr: ' + data.toString());
+        })
+        python.stdout.on('data', (data) => {
+          log('Python stdout: ' + data.toString());
+        })
+        res.send('true');
+        return;
+      }
+      else if (action === 'runprod') {
+
+        /* The 'Import to Production' button was clicked */
+        await importToProduction();
+        res.send('true');
+        return;
+      }
+      else {
+        console.log(req.user);
+        res.render('dashboard.ejs', { userData: req.user });
+        return;
+      }
+
+    } catch (e) {
+      return next(e);
+    }
+  });
+
+  /* API method to pull logs from the public.etl_run_log table */
+  app.get("/admin/dashboard/etl-log", async (req, res, next) => {
+    try {
+
+      let log = null;
+
+      await pool.query('select * from etl_run_log order by time_stamp asc;', async (err, result) => {
```

Comment:
Would be good to keep query case/syntax consistent.  Here the query is lower case but in other routes it is upper.

Reply:
Fixed